### ISSUE TITLE
CakePHP 5 compatibility

### ISSUE THREAD

**Issue by LordSimal** [passchn/cakephp-assets#11]
[CakePHP 5 RC1](https://github.com/cakephp/cakephp/releases/tag/5.0.0-RC1) just relased so will this plugin add support for that version?

**Comment by passchn** [comment]
The plugin depends on `josegonzalez/cakephp-upload` which does currently not support cake5: https://github.com/FriendsOfCake/cakephp-upload/blob/cake-5/composer.json 

When trying to upgrade my plugins and cake to version 5, I run into this error `composer require --update-with-dependencies -W "cakephp/cakephp:5.x-dev" "dereuromark/cakephp-queue:cake5-dev" "dereuromark/cakephp-tools:cake5-dev" "friendsofcake/search:cake-5-dev" "cakedc/cakephp-phpstan:3.next-cake5-dev" "cakephp/bake:3.x-dev" "cakephp/debug_kit:5.x-dev" "dereuromark/cakephp-ide-helper:cake5-dev" "cakephp/migrations:4.x-dev" "josegonzalez/cakephp-upload:cake-5-dev"`: 

```
Problem 3
    - Root composer.json requires cakephp/cakephp 5.x-dev -> satisfiable by cakephp/cakephp[5.x-dev].
    - josegonzalez/cakephp-upload dev-cake-5 requires cakephp/orm ^4.0.2 -> satisfiable by cakephp/orm[4.0.2, ..., 4.x-dev].
    - Only one of these can be installed: cakephp/orm[4.0.2, ..., 4.x-dev, 5.x-dev], cakephp/cakephp[5.x-dev]. cakephp/cakephp replaces cakephp/orm and thus cannot coexist with it.
    - Root composer.json requires josegonzalez/cakephp-upload cake-5-dev -> satisfiable by josegonzalez/cakephp-upload[dev-cake-5].
```

**Comment by passchn** [comment]
I created a cake5 branch for this plugin and my vite plugin, but the assets plugin is currently not composer-installable due to the  conflict with cakephp/orm: 
https://github.com/brandcom/cakephp-assets/commit/dcf27b3e6a277c74337f286221164c8808d1686a

**Comment by passchn** [comment]
See also: https://github.com/FriendsOfCake/cakephp-upload/issues/595

**Comment by passchn** [comment]
I now released version 0.5 which supports CakePHP 5. Didn't want to do a major release yet. 
https://github.com/brandcom/cakephp-assets/releases/tag/v0.5.0 

I also added a version map here: 
https://github.com/brandcom/cakephp-assets#cakephp-version-map 

The unit tests do not work currently as constants, e.g. `WWW_ROOT` cannot be found. This seems to be due to the fact that global functions are now opt-in: https://book.cakephp.org/5/en/appendices/5-0-migration-guide.html#global 

Also, I'm not sure if the testing setup is correct after the upgrade to phpunit 10.

**Comment by passchn** [comment]
Open tasks to remove `"minimum-stability": "dev"` in composer.json: 

- [x] install `josegonzalez/cakephp-upload` at a stable version
- [x] install `cakedc/cakephp-phpstan` at a stable version

**Comment by passchn** [comment]
CI-Fixes from @LordSimal were released in v0.5.1: 
https://github.com/brandcom/cakephp-assets/pull/12 
https://github.com/brandcom/cakephp-assets/releases/tag/v0.5.1

**Comment by passchn** [comment]
A fully-stable release without dev-versions is now available as v0.5.2:
https://github.com/brandcom/cakephp-assets/releases/tag/v0.5.2
